### PR TITLE
Improve show/hide reference button in side-by-side view

### PR DESCRIPTION
**Pull request by jreys**
#### Summary of Changes

See commits
#### Testing Instructions

Test if save target is showing when switching languages, test show/hide reference button.

### CODE REVIEW

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 1]
```diff
@@ -99,10 +114,11 @@
 
 	// Attach behaviour to target frame load event.
 	$('#target-association').on('load', function() {
-
 		// We need to check if we are not loading a blank iframe.
-		if (this.src != '')
+		if ($(this).attr('src') != '')
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: why adding _jquery_ when we can do it with _vanilla js_?

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 2]
```diff
@@ -51,7 +52,20 @@
 	Joomla.loadingLayer('load');
 
 	// Attach behaviour to toggle button.
-	$('body').on('click', '#toogle-left-panel', function() {
+	$('body').on('click', '#toogle-left-panel', function()
+	{
+		referenceHide = $('#adminForm').attr('data-hide-reference');
+		referenceShow = $('#adminForm').attr('data-show-reference');
+
+		if ($('#toogle-left-panel').text() === referenceHide)
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: _$(this)_ ?

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 3]
```diff
@@ -51,7 +52,20 @@
...
+		{
+			$('#toogle-left-panel').text(referenceShow);
+		}
+		else
+		{
+			$('#toogle-left-panel').text(referenceHide);
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: _$(this)_ ?

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 4]
```diff
@@ -51,7 +52,20 @@
...
 	// Attach behaviour to toggle button.
-	$('body').on('click', '#toogle-left-panel', function() {
+	$('body').on('click', '#toogle-left-panel', function()
+	{
+		referenceHide = $('#adminForm').attr('data-hide-reference');
+		referenceShow = $('#adminForm').attr('data-show-reference');
+
+		if ($('#toogle-left-panel').text() === referenceHide)
+		{
+			$('#toogle-left-panel').text(referenceShow);
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: _$(this)_ ?

File: administrator/components/com_associations/language/en-GB/en-GB.com_associations.ini [hunk 5]
```diff
@@ -9,6 +9,8 @@ COM_ASSOCIATIONS_COMPONENT_SELECTOR_DESC="Select a component from this list"
 COM_ASSOCIATIONS_COMPONENT_SELECTOR_LABEL="Select component"
 COM_ASSOCIATIONS_CONFIGURATION="Multilingual Associations: Options"
 COM_ASSOCIATIONS_COPY_REFERENCE="Copy Reference to Target"
+COM_ASSOCIATIONS_EDIT_SHOW_REFERENCE="Show Reference"
+COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE="Hide Reference"
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/language/en-GB/en-GB.com_associations.ini:12]: alpha order the language strings

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 6]
```diff
@@ -235,13 +251,15 @@
...
+<button id="toogle-left-panel" class="btn btn-small"><?php echo JText::_('COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE'); ?></button>
 
 <form action="<?php echo JRoute::_(
 			'index.php?option=com_associations&view=association&layout=' . $layout . '&acomponent='
 			. $aComponent . '&aview=' . $aView . '&extension=' . $extension . '&referencelanguage=' . $rLanguage . '&id='
 			. $this->referenceId
-		); ?>" method="post" name="adminForm" id="adminForm" class="form-validate" data-associatedview="<?php echo $this->associatedView; ?>">
+		); ?>" method="post" name="adminForm" id="adminForm" class="form-validate" data-associatedview="<?php echo $this->associatedView; ?>"
+		data-show-reference="<?php echo JText::_('COM_ASSOCIATIONS_EDIT_SHOW_REFERENCE'); ?>"
+		data-hide-reference="<?php echo JText::_('COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE'); ?>">
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: please move the data attributes to the element that uses it, ie, the button itself.

data-\* elements are always related to the html element itself.
So if this is text of the button it should be in the button html tag.

example:

``` html
<button id="toogle-left-panel" class="btn btn-small"
data-show-reference="<?php echo JText::_('COM_ASSOCIATIONS_EDIT_SHOW_REFERENCE'); ?>"
data-hide-reference="<?php echo JText::_('COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE'); ?>">...</button>
```

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 7]
```diff
@@ -251,15 +251,16 @@
 $extension  = $input->get('extension', '', 'string');
 $rLanguage  = $input->get('referencelanguage', '', 'string') != null ? $input->get('referencelanguage', '', 'string') : '';
 ?>
-<button id="toogle-left-panel" class="btn btn-small"><?php echo JText::_('COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE'); ?></button>
+<button id="toogle-left-panel" class="btn btn-small"><?php echo JText::_('COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE'); 
+?>  	data-show-reference="<?php echo JText::_('COM_ASSOCIATIONS_EDIT_SHOW_REFERENCE'); ?>"
+		data-hide-reference="<?php echo JText::_('COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE'); ?>"
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: where is closing tag and the default button text?

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 8]
```diff
@@ -54,16 +54,16 @@
 	// Attach behaviour to toggle button.
 	$('body').on('click', '#toogle-left-panel', function()
 	{
-		referenceHide = $('#adminForm').attr('data-hide-reference');
-		referenceShow = $('#adminForm').attr('data-show-reference');
+		referenceHide = $(this).attr('data-hide-reference');
+		referenceShow = $(this).attr('data-show-reference');
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: vannila js
this.getAttribute('data-hide-reference');
this.getAttribute('data-show-reference');

File: administrator/components/com_associations/views/association/tmpl/edit.php (hunk 7 above)
Comment (jreys) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: the closing tag is below and what do you mean by default button text?

``` php
<?php echo JText::_('COM_ASSOCIATIONS_EDIT_HIDE_REFERENCE'); ?>
```

Comment (jreys) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: nvm, my mistake, fixing

File: administrator/components/com_associations/views/association/tmpl/edit.php [hunk 9]
```diff
@@ -51,7 +52,20 @@
 	Joomla.loadingLayer('load');
 
 	// Attach behaviour to toggle button.
-	$('body').on('click', '#toogle-left-panel', function() {
+	$('body').on('click', '#toogle-left-panel', function()
+	{
+		referenceHide = $(this).attr('data-hide-reference');
+		referenceShow = $(this).attr('data-show-reference');
```
Comment (andrepereiradasilva) [inline administrator/components/com_associations/views/association/tmpl/edit.php]: vanilla js...
var referenceHide = this.getAttribute('data-hide-reference');
var referenceShow = this.getAttribute('data-show-reference');

**Comment by andrepereiradasilva** [comment]
i tested and this is all ok 

Simple changes to complete a incomplete button.

will merge

**Pull request merged**